### PR TITLE
audout: Implement and fix some calls

### LINKED ISSUE #1133: audio IAudioOut: 9 (GetAudioOutBufferCount) is not implemented
audio IAudioOut: 9 (GetAudioOutBufferCount) is not implemented.

[Switchbrew](https://switchbrew.org/wiki/Audio_services#GetAudioOutBufferCount) | [SwIPC](https://reswitched.github.io/SwIPC/ifaces.html#nn::audio::detail::IAudioOut)

## Service description:
```c
// Returns u32 amount of buffers awaiting playback.
GetAudioOutBufferCount() -> u32 count
```

## Required by:
* https://github.com/Ryujinx/Ryujinx-Games-List/issues/822

### LINKED ISSUE #1258: audio IAudioOut: 10 (GetAudioOutPlayedSampleCount) is not implemented
audio IAudioOut: 10 (GetAudioOutPlayedSampleCount) is not implemented

[SwitchBrew](https://switchbrew.org/wiki/Audio_services#GetAudioOutPlayedSampleCount) | [SwIPC](https://reswitched.github.io/SwIPC/ifaces.html#nn::audio::detail::IAudioOut(10))
## Service description
```
GetAudioOutPlayedSampleCount

Returns u64 total amount of samples played back. 
```
## Required by
Ryujinx/Ryujinx-Games-List/issues/1262

### LINKED ISSUE #1519: audout:u IAudioOut: 11 (FlushAudioOutBuffers) is not implemented
### What service call is missing?

audout:u IAudioOut: 11 (FlushAudioOutBuffers) is not implemented

[Switchbrew](https://switchbrew.org/wiki/Audio_services#FlushAudioOutBuffers) | [Reswitched](https://reswitched.github.io/SwIPC/ifaces.html#nn::audio::detail::IAudioOut(11))

### Service description

```
Flushes all pending buffers. Returns 1 if buffers were pending.
```

### Required by: 

https://github.com/Ryujinx/Ryujinx-Games-List/issues/2468

### PR DESCRIPTION
**Pull request by AcK77**
This PR implement audout service calls:
- GetAudioOutBufferCount
- GetAudioOutPlayedSampleCount
- FlushAudioOutBuffers

The RE of calls just give some hints about no extra checks has done.
Since we use a totally different implementation because of our backend, I can't do something better for now.

SetAudioOutVolume and GetAudioOutVolume are fixed too by set/get the volume of the current opened track, previous implementation was wrong.

This fix #1133, fix #1258 and fix #1519.

Thanks to @jduncanator for his help during the implementation and all his precious advices.

Now some games goes ingame and are fully playable, I haven't tested all btw.
image
Capture

### CODE REVIEW

File: Ryujinx.Audio/Renderers/OpenAL/OpenALAudioTrack.cs [hunk 1]
```diff
@@ -11,9 +11,12 @@ internal class OpenALAudioTrack : IDisposable
         public int           SampleRate { get; private set; }
         public ALFormat      Format     { get; private set; }
         public PlaybackState State      { get; set; }
+        public float         Volume     { get; private set; }
 
         public int HardwareChannels { get; }
         public int VirtualChannels { get; }
+        public uint BufferCount => (uint)_buffers.Count;
+        public ulong PlayedSampleCount { get; set; }
```
Comment (jduncanator) [inline Ryujinx.Audio/Renderers/OpenAL/OpenALAudioTrack.cs:19]: This should be `private set;`

File: Ryujinx.Audio/Renderers/SoundIo/SoundIoAudioTrackPool.cs [hunk 2]
```diff
@@ -174,6 +174,75 @@ public void Put(SoundIoAudioTrack track)
             m_Queue.Enqueue(track);
         }
 
+        /// <summary>
```
Comment (jduncanator) [inline Ryujinx.Audio/Renderers/SoundIo/SoundIoAudioTrackPool.cs:177]: These functions shouldn't live in the TrackPool. The TrackPool's job is to simply provide and count Track objects, much like an ArrayPool. These functions should live in SoundIoAudioOut.

**Review by jduncanator — COMMENTED** [review]
Looks good to me, nice work! I don't think the methods should live in the `SoundIoAudioTrackPool` class however. This is a simple class that's sole job is to issue `SoundIoAudioTrack` objects to the `SoundIoAudioOut` manager. They should live in the `SoundIoAudioOut` class.

Other than that, I'm not really an expert on the OpenAL backend, so can only wager a guess that it _looks_ correct. I'm assuming proper testing is done on both backends 👍 

Cheers!

File: Ryujinx.Audio/Renderers/OpenAL/OpenALAudioTrack.cs (hunk 1 above)
Comment (jduncanator) [reply Ryujinx.Audio/Renderers/OpenAL/OpenALAudioTrack.cs:19]: Nevermind, it's incremented in the OpenAaAudioOut due to the architecture of the implementation 😬

**Review by jduncanator — APPROVED** [review]
Looks good to me, thanks 👍

**Review by Thog — APPROVED** [review]

**Pull request merged**